File: src/backend-logi/defake-logi-backend/src/main/java/com/group6/defakelogibackend/model/Notification.java
package com.group6.defakelogibackend.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Notification {
    private long id;
    private long templateId;
    private long userIdSent;
    private long userIdRec;
    private long groupId;
    private String title;
    private String content;
    private LocalDateTime sentAt;
    private LocalDateTime readAt;

}

File: src/frontend/src/api/notification.ts
import { http } from "@/utils/http";
export interface Notification {
  id: number;
  templateId: number;
  userIdSent: number;
  userIdRec: number;
  groupId: number;
  title: string;
  content: string;
  sentAt: string;
  readAt: string | null;
}

export interface GetAllNotificationResult {
  code: number;
  message: string;
  data: Notification[];
}

export interface ReadNotificationResult {
  code: number;
  message: string;
  data?: any;
}


/** 获取通知 */
export const getAllNotification = (data: { condition: string | null }) => {
  if (data.condition === null) {
    return http.request<GetAllNotificationResult>(
      "get",
      "/api/notification/filter",
      {
        params: {
          condition: "sent_at_asc"
        }
      }
    )
  }
  return http.request<GetAllNotificationResult>(
    "get",
    "/api/notification/filter",
    {
      params: data
    }
  );
};

/** 阅读通知 */
export const readNotification = (data: { notificationId: string }) => {
  return http.request<ReadNotificationResult>(
    "put",
    "/api/notification/read",
    { data }
  );
};